spec/avram/queryable_spec.cr

```diff
@@ -856,6 +856,95 @@ describe Avram::Queryable do
     end
   end
 
+  describe "#group_count" do
+    context "when no records exist" do
+      it "0 grouped columns returns { [] => 0 }" do
+        query = UserQuery.new
+
+        query.group_count.should eq({ [] of String => 0 })
+      end
+
+      it "1 grouped columns returns { [] => 0 }" do # TODO: What should we do here?
+        query = UserQuery.new.group(&.name)
+
+        query.group_count.should eq({ [] of String => 0 })
+      end
+
+      it "2 grouped columns returns { [] => 0 }" do # TODO: What should we do here?
+        query = UserQuery.new.group(&.name)
+
+        query.group_count.should eq({ [] of String => 0 })
+      end
+    end
+
+    context "when 1 record exists" do
+      before_each do
+        UserFactory.create do |user|
+          user.age(32)
+          user.name("Taylor")
+        end
+      end
+
+      it "0 grouped columns returns { [] => 1 }" do
+        query = UserQuery.new
+
+        query.group_count.should eq({ [] of String => 1 })
+      end
+
+      it "1 grouped columns (age) returns { [42] => 1 }" do
+        query = UserQuery.new.group &.age
+
+        query.group_count.should eq({ [32] => 1 })
+      end
+
+      it "1 grouped columns (age, name) returns { [42, \"Gumby\"] => 1 }" do
+
+        query = UserQuery.new.group(&.age).group(&.name)
+
+        query.group_count.should eq({ [32, "Taylor"] => 1 })
+      end
+    end
+
+    context "when matrix [32, Daniel] [32, Taylor] [32, Taylor] [44, Shakira]" do
```

Comment:
All of these tests might be overkill or maybe they're not enough, I'm not really sure.
I tried to express the various returns that could happen with `zero | one | many` groups